golink.go

```diff
@@ -541,24 +544,42 @@ func expandLink(long string, env expandEnv) (*url.URL, error) {
 
 func devMode() bool { return *dev != "" }
 
+const peerCapName = "tailscale.com/golink"
+
+type capabilities struct {
+	Admin bool `json:"admin"`
+}
+
+type user struct {
+	login   string
+	isAdmin bool
+}
+
 // currentUser returns the Tailscale user associated with the request.
 // In most cases, this will be the user that owns the device that made the request.
 // For tagged devices, the value "tagged-devices" is returned.
 // If the user can't be determined (such as requests coming through a subnet router),
 // an error is returned unless the -allow-unknown-users flag is set.
-var currentUser = func(r *http.Request) (string, error) {
+var currentUser = func(r *http.Request) (u user, err error) {
 	if devMode() {
-		return "foo@example.com", nil
+		return user{login: "foo@example.com"}, nil
 	}
 	whois, err := localClient.WhoIs(r.Context(), r.RemoteAddr)
 	if err != nil {
 		if *allowUnknownUsers {
 			// Don't report the error if we are allowing unknown users.
-			return "", nil
+			return u, nil
+		}
+		return u, err
+	}
+	u.login = whois.UserProfile.LoginName
+	caps, _ := tailcfg.UnmarshalCapJSON[capabilities](whois.CapMap, peerCapName)
+	for _, cap := range caps {
+		if cap.Admin {
+			u.isAdmin = true
```

Comment:
Could we not at this point return directly? Then we don't really need named results:

```go
login := whois.UserProfile.LoginName
// ...
for _, cap := range caps {
   if cap.Admin {
      return user{login: login, isAdmin: true}, nil
   }
}
return user{login: login}, nil
```

or words to that effect?

Reply:
done